src/main/java/org/apache/gossip/manager/GossipManager.java

```diff
@@ -207,11 +208,12 @@ public void shutdown() {
     gossipServiceRunning.set(false);
     gossipThreadExecutor.shutdown();
     gossipCore.shutdown();
-    if (passiveGossipThread != null) {
-      passiveGossipThread.shutdown();
-    }
-    if (activeGossipThread != null) {
-      activeGossipThread.shutdown();
+    if(null != this.bundle) {
```

Comment:
Not very sure about this. Does it mean changing `if(null != this.bundle)` to `if(null != bundle)` is enough or reverting back to use 
`
  if (passiveGossipThread != null) {
    passiveGossipThread.shutdown();
  }
  if (activeGossipThread != null) {
    activeGossipThread.shutdown();
  }
`
shutting down threads individually?